### PR TITLE
feature: upstream support websocket enable

### PR DESCRIPTION
**Pull request by lilien1010**
# Summary

 change websocket enable feature into upstream 

### Full changelog

* lua/apisix.lua
* lua/apisix/schema_def.lua
* t/lib/server.lua
* t/node/upstream-websocket.t

### Issues resolved

Fix https://github.com/apache/incubator-apisix/issues/1050

### CODE REVIEW

File: lua/apisix/schema_def.lua [hunk 1]
```diff
@@ -273,6 +273,11 @@ local upstream_schema = {
             description = "the key of chash for dynamic load balancing",
             type = "string",
         },
+        enable_websocket = {
+            description = "enable websocket for request",
+            type        = "boolean",
+            default     = false
```
Comment (membphis) [inline lua/apisix/schema_def.lua]: we can delete this `default`, we can use `nil` as `false`

File: doc/plugins/proxy-rewrite-cn.md [hunk 2]
```diff
@@ -29,7 +29,6 @@
 |uri             |可选| 转发到上游的新 `uri` 地址|
 |regex_uri       |可选| 转发到上游的新 `uri` 地址, 使用正则表达式匹配来自客户端的uri，当匹配成功后使用模板替换转发到上游的uri, 未匹配成功时将客户端请求的uri转发至上游。当`uri`和`regex_uri`同时存在时，`uri`优先被使用。例如：["^/iresty/(.*)/(.*)/(.*)","/$1-$2-$3"] 第一个元素代表匹配来自客户端请求的uri正则表达式，第二个元素代表匹配成功后转发到上游的uri模板|
 |host            |可选| 转发到上游的新 `host` 地址，例如：`iresty.com` |
-|enable_websocket|可选| 是否启用 `websocket`（布尔值），默认不启用|
```
Comment (moonming) [inline doc/plugins/proxy-rewrite-cn.md]: you not remove the code in `proxy-rewrite.lua`

Comment (membphis) [reply doc/plugins/proxy-rewrite-cn.md]: we can do this in a new separate PR

Comment (moonming) [reply doc/plugins/proxy-rewrite-cn.md]: please keep the doc and code are the same.

Comment (lilien1010) [reply doc/plugins/proxy-rewrite-cn.md]: sure, will revert proxy-write's doc and change it in another PR

File: lua/apisix/schema_def.lua (hunk 1 above)
Comment (lilien1010) [reply lua/apisix/schema_def.lua]: sure

File: lua/apisix/schema_def.lua [hunk 3]
```diff
@@ -276,7 +276,7 @@ local upstream_schema = {
         enable_websocket = {
             description = "enable websocket for request",
             type        = "boolean",
-            default     = false
+            default     = nil
```
Comment (membphis) [inline lua/apisix/schema_def.lua]: we can delete this line, it is useless ^_^

Comment (lilien1010) [reply lua/apisix/schema_def.lua]: ok

File: doc/architecture-design-cn.md [hunk 4]
```diff
@@ -241,6 +241,7 @@ APISIX 的 Upstream 除了基本的复杂均衡算法选择外，还支持对上
 |key             |必需|该选项只有 `type` 是 `chash` 才有效，需要配合 `hash_on` 来使用，通过 `hash_on` 和 `key` 来查找对应的 node `id`。`hash_on` 设为 `vars` 时，`key` 为必传参数，目前支持的 Nginx 内置变量有 `uri, server_name, server_addr, request_uri, remote_port, remote_addr, query_string, host, hostname, arg_***`，其中 `arg_***` 是来自URL的请求参数，[Nginx 变量列表](http://nginx.org/en/docs/varindex.html)；`hash_on` 设为 `header` 时, `key` 为必传参数，自定义的 `header name`；`hash_on` 设为 `cookie` 时, `key` 为必传参数， 自定义的 `cookie name`；`hash_on` 设为 `consumer` 时，`key` 不需要设置，为空，此时哈希算法采用的 `key` 为认证通过的 `consumer_id`。 如果指定的 `hash_on` 和 `key` 获取不到值时，默认取 `remote_addr`|
 |checks          |可选|配置健康检查的参数，详细可参考[health-check](health-check.md)|
 |retries         |可选|使用底层的 Nginx 重试机制将请求传递给下一个上游，默认 APISIX 会启用重试机制，根据配置的后端节点个数设置重试次数，如果此参数显式被设置将会覆盖系统默认设置的重试次数。|
+|enable_websocket|可选| 是否启用 `websocket`（布尔值），默认不启用|
```
Comment (moonming) [inline doc/architecture-design-cn.md]: no test case to cover the default `false`

Comment (membphis) [reply doc/architecture-design-cn.md]: I think the old test cases should have covered this case. We can merge this PR first.

Comment (lilien1010) [reply doc/architecture-design-cn.md]: already added in `upstream-websocket.t`
=== TEST 6: disable websocket(id: 6)

Comment (membphis) [reply doc/architecture-design-cn.md]: got it

**Pull request merged**